File: domain/src/main/java/no/nav/melosys/domain/eessi/sed/Arbeidssted.java
package no.nav.melosys.domain.eessi.sed;

import no.nav.melosys.domain.mottatteopplysninger.data.arbeidssteder.FysiskArbeidssted;
import org.apache.commons.lang3.StringUtils;

public class Arbeidssted {
    private static final String IKKE_TILGJENGELIG = "N/A";

    private String navn;
    private Adresse adresse;
    private boolean fysisk;
    private String hjemmebase;

    public static Arbeidssted lagIkkeFastArbeidssted(String landkode) {
        Arbeidssted arbeidssted = new Arbeidssted();
        arbeidssted.setNavn(Adresse.INGEN_FAST_ADRESSE);
        arbeidssted.setAdresse(Adresse.lagIkkeFastAdresse(landkode));
        return arbeidssted;
    }

    public FysiskArbeidssted tilFysiskArbeidssted() {
        FysiskArbeidssted fysiskArbeidssted = new FysiskArbeidssted();

        fysiskArbeidssted.setAdresse(adresse.tilStrukturertAdresse());
        fysiskArbeidssted.setVirksomhetNavn(navn);

        return fysiskArbeidssted;
    }

    public String getNavn() {
        return navn;
    }

    public void setNavn(String navn) {
        this.navn = StringUtils.isBlank(navn) ? IKKE_TILGJENGELIG : navn;
    }

    public Adresse getAdresse() {
        return adresse;
    }

    public void setAdresse(Adresse adresse) {
        this.adresse = adresse;
    }

    public boolean isFysisk() {
        return fysisk;
    }

    public void setFysisk(boolean fysisk) {
        this.fysisk = fysisk;
    }

    public String getHjemmebase() {
        return hjemmebase;
    }

    public void setHjemmebase(String hjemmebase) {
        this.hjemmebase = hjemmebase;
    }
}

File: domain/src/main/kotlin/no/nav/melosys/domain/eessi/sed/Arbeidssted.kt
package no.nav.melosys.domain.eessi.sed

import no.nav.melosys.domain.mottatteopplysninger.data.arbeidssteder.FysiskArbeidssted
import org.apache.commons.lang3.StringUtils

class Arbeidssted {
    var navn: String? = null
        set(value) {
            field = if (StringUtils.isBlank(value)) IKKE_TILGJENGELIG else value
        }

    var adresse: Adresse? = null

    var fysisk: Boolean = false

    var hjemmebase: String? = null

    fun tilFysiskArbeidssted(): FysiskArbeidssted =
        FysiskArbeidssted(navn, adresse?.tilStrukturertAdresse() ?: throw IllegalStateException("Adresse cannot be null"))

    companion object {
        private const val IKKE_TILGJENGELIG = "N/A"

        @JvmStatic
        fun lagIkkeFastArbeidssted(landkode: String) = Arbeidssted().apply {
            navn = Adresse.INGEN_FAST_ADRESSE
            adresse = Adresse.lagIkkeFastAdresse(landkode)
        }
    }
}
